### PR TITLE
Wrap "unknown" group creation in safe upsert

### PR DESCRIPTION
**Pull request by nagem**
Ported from #966.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/dao/dbutil.py [hunk 1]
```diff
@@ -43,6 +43,19 @@ def fault_tolerant_replace_one(db, coll_name, query, update, upsert=False):
...
+    https://jira.mongodb.org/browse/SERVER-14322
+    This function will try an update_one operation, returning the result and if the operation succeeded.
+    """
+    try:
+        result = db[coll_name].update_one(query, update, upsert=upsert)
+    except DuplicateKeyError:
+        return result, False
+    else:
+        return result, True
+
```
Comment (kofalt) [inline api/dao/dbutil.py]: Suggest throwing this above `fault_tolerant_replace_one` to live with `try_replace_one`.

File: api/dao/dbutil.py [hunk 2]
```diff
@@ -43,6 +43,19 @@ def fault_tolerant_replace_one(db, coll_name, query, update, upsert=False):
 
     raise APIStorageException('Unable to replace object.')
 
+def try_update_one(db, coll_name, query, update, upsert=False):
+    """
+    Mongo does not see update w/ upsert as an atomic action:
+    https://jira.mongodb.org/browse/SERVER-14322
```
Comment (kofalt) [inline api/dao/dbutil.py]: May as well put this bit above `try_replace_one` and say it applies to the `try_` family

File: api/config.py [hunk 3]
```diff
@@ -198,7 +198,12 @@ def initialize_db():
     create_or_recreate_ttl_index('job_tickets', 'timestamp', 3600) # IMPORTANT: this controls job orphan logic. Ref queue.py
 
     now = datetime.datetime.utcnow()
-    db.groups.update_one({'_id': 'unknown'}, {'$setOnInsert': { 'created': now, 'modified': now, 'label': 'Unknown', 'permissions': []}}, upsert=True)
+    try_update_one(db,
+                   'groups', {'_id': 'unknown'},
+                   {'$setOnInsert': {'created': now, 'modified': now, 'label': 'Unknown', 'permissions': []}},
+                   upsert=True)
+    log.info('Initializing database, creating indexes ....DONE')
```
Comment (kofalt) [inline api/config.py]: Suggest removing the log line. We have enough startup spam & we won't programatically trigger on it.

File: api/dao/dbutil.py (hunk 1 above)
Comment (nagem) [reply api/dao/dbutil.py]: I created family lines around the mongo action instead.

**Pull request merged**